smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/WaiterGenerator.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.waiters
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftDelegator
+import software.amazon.smithy.swift.codegen.SwiftDependency
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+import software.amazon.smithy.waiters.WaitableTrait
+
+/**
+ * Generate waiters for supporting operations.  See
+ * https://smithy.io/2.0/additional-specs/waiters.html for details.
+ */
+class WaiterGenerator : SwiftIntegration {
+    override fun enabledForService(model: Model, settings: SwiftSettings): Boolean {
+        return serviceHasWaiters(model)
+    }
+
+    override fun writeAdditionalFiles(
+        ctx: CodegenContext,
+        protoCtx: ProtocolGenerator.GenerationContext,
+        delegator: SwiftDelegator
+    ) {
+        if (!serviceHasWaiters(ctx.model)) return
+        val service = ctx.model.expectShape<ServiceShape>(ctx.settings.service)
+
+        // Open a new file Waiters.swift to hold the waiter definitions for this service
+        val waiterFilename = "${ctx.settings.moduleName}/Waiters.swift"
+        delegator.useFileWriter(waiterFilename) { writer ->
+            writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
+            val serviceSymbol = ctx.symbolProvider.toSymbol(service)
+
+            // Only add waiters if Swift version is 5.7 or higher
+            writer.write("#if swift(>=5.7)")
```

Comment:
Can we add a comment here describing why we only support waiters for 5.7 or higher

Reply:
This is removed, since explicit types have been added to make acceptors compile on Swift 5.5+